src/common/types/http/http-options.ts

```diff
@@ -6,6 +6,8 @@ type HttpOptions = {
   payload: BodyInit_ | null;
   hasAuth: boolean;
   query: Record<string, unknown>;
+  notifyError: boolean;
+  customErrorMessage: string;
```

Comment:
Maybe it's better to combine these properties into one config like this:
```
  type HttpErrorConfig = {
    notify?: boolean;
    message?: string;
  }
```

```suggestion
  errorConfig: HttpErrorConfig;
```